File: Sagittarius-StarterKits/Java/sagittarius/SagRequest.java
/**
 * Sagittarius - Java Starter Kit
 * Copyright 2013 WillyG Productions
 * @Authors: William Gaul
 */
package sagittarius;

public class SagRequest {

    private String dest, data, mID, qID;
    private Sagittarius parent;
    private String delim;

    public SagRequest(Sagittarius s) {
        this.parent = s;
        this.data = "";
        this.delim = "";
    }
    
    public void submit() {
        parent.SubmitRequest(this);
    }
    
    public void addURLPair(String param, String value, boolean encryptValue) {
        if (encryptValue) {
            value = parent.encrypt(value);
        }
        this.data += (delim + param + "=" + value);
        this.delim = "&";
    }
    
    public SagRequest setDestination(String dest) {
        this.dest = dest;
        return this;
    }
    
    public SagRequest setModuleInfo(String mID, String qID) {
        this.mID = mID;
        this.qID = qID;
        return this;
    }

    public String getDestination() {
        return dest;
    }

    public String getData() {
        return data;
    }

    public String getModuleID() {
        return mID;
    }

    public String getQueryID() {
        return qID;
    }
}

File: starter-kits/javascript/src/sag-request.js
/**
 * Sagittarius - JavaScript Starter Kit
 * Copyright WillyG Productions
 * @Authors: William Gaul
 */
var encryption = require('./encryption');

module.exports = function(password) {
	var that = {}, data = [];
	that.password = password;

	that.setDestination = function(dest) {
		that.dest = dest;
		return that;
	}

	that.addURLPair = function(key, value, encrypt) {
		if (encrypt !== undefined) {
			value = encryption.encrypt(value, that.password);
		}
		data.push({key: key, value: value});
	};
	
	that.setURLPair = function(key, value, encrypt) {
		for (var i = 0; i < data.length; i++) {
			if (data[i].key === key) {
				if (encrypt !== undefined) {
					value = encryption.encrypt(value, that.password);
				}
				data[i].value = value;
				return;
			}
		}
	};

	that.getData = function() {
		var params = {};
		data.forEach(function(param) {
			params[param.key] = param.value;
		});
		return params;
	};

	return that;
};
